android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/EngineObserver.kt

```diff
@@ -82,7 +82,11 @@ internal class EngineObserver(
         store.dispatch(ContentAction.UpdateFirstContentfulPaintStateAction(tabId, false))
     }
 
-    override fun onLocationChange(url: String) {
+    override fun onLocationChange(url: String, hasUserGesture: Boolean) {
+        if (hasUserGesture) {
+            store.dispatch(ContentAction.UpdateSearchTermsAction(tabId, ""))
```

Comment:
This is considered a side-effect. We should pass the `hasUserGesture` in the `UpdateUrlAction` and let a middleware (existing probably) dispatch this action.